src/app/container/container.component.ts

```diff
@@ -197,8 +200,11 @@ export class ContainerComponent extends Entry implements AfterViewInit {
     if (tool) {
       this.tool = tool;
       this.initTool();
+      this.userQuery.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((userData) => {
+        this.username = userData.username ? userData.username : 'not-logged-in';
```

Comment:
1. Doesn't this mean that if you're not logged in, an email will be sent saying "My user name... is not-logged-in"? I guess you have a chance to edit the email before sending it though, don't you?
2. The message in email.service.ts says "My user name on ${toolRegistry} is ${username}", but you're passing the Dockstore username, not the toolRegistry username.

Reply:
1. Yes you will be able to edit the email before sent!
2. Oh no sorry I had a misunderstanding...have no idea why I thought username on dockstore is the username used there. Now I have been trying to retrieve user information, but it seems like the tool registry username is not stored anywhere in `ExtendedDockstoreTool.ts` or `dockstoreTool.ts`. I'm wondering if you, or any others have any insights on this?(Should I ask this question in the channel?)

Reply:
A few things:
- The username should be the user's username from the Docker registry
- Dockstore may or may not have that username (if the account's not linked in Dockstore)
- If the account is not linked, there's not much we can do. User will have to fill it out.
- If we want to display the username for the accounts that are linked, it would likely involve getting the user's current tokens, hopefully finding the registry there, getting the username from the matching token.

IMO, it's not worth the effort.

What does everyone else think?

Reply:
I would think not worth it. At least for now. We only have 2 private tools.

Reply:
There's also the case of a user not even being logged in to Dockstore at all...

I don't think it's worth a lot of effort, and I think we could have a the message saying something like "<put your Docker Hub/quay.io username here>" (or put the actual name if it's easily accessible -- how does the accounts page do it?).

I don't think we should have a message that says "not-loggged-in" or has the wrong username.

Reply:
So I assume we can just display the message like <put your Docker Hub/quay.io username here> so the user can edit it themselves while sending the email request? If this is the case I will just modify the other PR since it's a very small fix. Then we can disregard this PR

Reply:
I'm OK with some version of that, although to be clear, you'll know if it's Docker Hub or quay.io, so the message should be specific. But let's see what Gary and Denis say.

Reply:
Yeah, Charles's suggestion works for me